rr_common/src/planner/distance_checker.cpp

```diff
@@ -148,65 +148,57 @@ std::tuple<bool, double> DistanceChecker::GetCollisionDistance(const Pose& pose)
     double half_width = (hitbox_.width_left + hitbox_.width_right) / 2.;
 
     int i = GetCacheIndex(pose.x, pose.y);
+    if (i < 0) {
+        return std::nullopt;
+    }
 
-    bool collision = false;
-    double min_dist = 1000;
-
-    if (i >= 0) {
-        CacheEntry& entry = cache_[i];
-
-        if (entry.nearest_point != nullptr) {
-            entry.might_hit_points.push_back(entry.nearest_point);
+    const CacheEntry& entry = cache_[i];
+
+    auto point_in_local_frame = [search_x, search_y, cos_th, sin_th](const PCLPoint& p) {
+        double offsetX = p.x - search_x;
+        double offsetY = p.y - search_y;
+        double x = cos_th * offsetX + sin_th * offsetY;
+        double y = -sin_th * offsetX + cos_th * offsetY;
+        return std::make_tuple(x, y);
+    };
```

Comment:
Is there a reason why you don't use a `tf::Transform` and transform the point that way?

Reply:
It's this way because I already had this code, and just refactored it